### PR TITLE
Add IPv4 pricing to UI

### PR DESCRIPTION
**Pull request by enescakir**
I made location based pricing helper on UI more generic at 247c987.

When you add `.location-based-price` class and `data-amount`, `data-resource-type`, `data-resource-family` data attributes to an input, it updates location based when changed.

I refactored checkbox component little bit to add this attributes to options.

Also I added a description to IPv4 input. It's totally open to any corrections.

### CODE REVIEW

**Comment by enescakir** [comment]
Screenshot 2023-08-08 at 20 41 55

File: views/vm/create.erb [hunk 1]
```diff
@@ -73,9 +73,13 @@
...
-                    label: "Enable Public IPv4",
-                    value: "1",
-                    selected: "1"
+                    label: "Public IPv4 Support (+<span class='enable-ip4-1-monthly-price'>-</span><span class='text-xs text-gray-500'>/mo</span>)",
+                    options: [
+                     ["1", "Enable Public IPv4", "location-based-price", {"data-amount" => "1", "data-resource-type" => "IPAddress", "data-resource-family" => "IPv4"}]
+                    ],
+                    selected: "1",
+                    description:
+                      "Assigning a public IPv4 address to virtual machine allows to connect it via IPv4 or to allow virtual machine to connect public internet via IPv4."
```
Comment (byucesoy) [inline views/vm/create.erb:82]: Should we mention that some resources you want to connect might not support communication from IPv6?

**Review by byucesoy — APPROVED** [review]

File: views/vm/create.erb (hunk 1 above)
Comment (enescakir) [reply views/vm/create.erb:82]: It was my intention. We can make it more clear. Do you have any suggestion?

Comment (byucesoy) [reply views/vm/create.erb:82]: No suggestion. I don't want to use a language like "You cannot..." also don't want to name the names :) but we need to communicate potential limitations. Maybe something like;

> Without an IPv4 address, some web sites might not work properly if they are not supporting IPv6.

One thing to consider is to use something other than "web sites" as the internet is not just consist of web sites, but I'm not sure what would be a better terminology.

Comment (enescakir) [reply views/vm/create.erb:82]: @furkansahin Do you have any comment for it?

Comment (furkansahin) [reply views/vm/create.erb:82]: we can just say;
```
must have to access ipv4 only networks, such as, github.com.
```

Comment (enescakir) [reply views/vm/create.erb]: I updated as "Public IPv4 address is required to access IPv4 only networks, such as github.com."
Screenshot 2023-08-09 at 16 13 56

Comment (byucesoy) [reply views/vm/create.erb]: Do we want to specifically mention github.com?

Comment (enescakir) [reply views/vm/create.erb]: Lots of people think GitHub is the biggest barrier to switch only-IPv6 machines. But not sure to mention or not

Comment (enescakir) [reply views/vm/create.erb]: It mixed all feedbacks. What about this one?
> Needed for inbound and outbound public IPv4 connections. Websites that do not support IPv6 will be inaccessible without an IPv4 address.

**Pull request merged**